q2_types/feature_table/_format.py

```diff
@@ -33,5 +33,55 @@ def sniff(self):
             return False
 
 
-FeatureTableDirectoryFormat = model.SingleFileDirectoryFormat(
-    'FeatureTableDirectoryFormat', 'feature-table.biom', BIOMV1Format)
+class BIOMV210Format(model.BinaryFileFormat):
+    # minimum requirements as described by
+    # http://biom-format.org/documentation/format_versions/biom-2.1.html
+    groups = {'sample',
+              'sample/matrix',
+              'sample/metadata',
+              'sample/group-metadata',
+              'observation',
+              'observation/matrix',
+              'observation/metadata',
+              'observation/group-metadata'}
+    datasets = {'sample/ids',
+                'sample/matrix/data',
+                'sample/matrix/indptr',
+                'sample/matrix/indices',
+                'observation/ids',
+                'observation/matrix/data',
+                'observation/matrix/indptr',
+                'observation/matrix/indices'}
+    attrs = {'id',
+             'type',
+             'format-url',
+             'format-version',
+             'generated-by',
+             'creation-date',
+             'shape',
+             'nnz'}
+
+    def open(self):
+        return h5py.File(self.path, mode=self._mode)
```

Comment:
`self.path` is a python `pathlib.Path` object, not a string of the path. You probably want to pass in `str(self)` here instead.